c7n/resources/s3.py

```diff
@@ -2662,28 +2662,24 @@ class SetBucketEncryption(BucketActionBase):
         }
     }
 
-    permissions = ('s3:PutEncryptionConfiguration',
-                's3:GetEncryptionConfiguration',
-                'kms:ListAliases')
+    permissions = ('s3:PutEncryptionConfiguration', 's3:GetEncryptionConfiguration', 'kms:ListAliases')
 
     def process(self, buckets):
         keys = {}
-        regions = set()
         regions = {get_region(b) for b in buckets}
         key = self.data.get('key')
 
         if self.data.get('enabled', True) and key:
             keys = self.resolve_keys(regions, key)
 
         with self.executor_factory(max_workers=3) as w:
-            results = []
-            futures = {w.submit(self.process_bucket, b, keys) : b for b in buckets}
+            futures = {w.submit(self.process_bucket, b, keys): b for b in buckets}
             for future in as_completed(futures):
-                if future.exception():
-                    bucket = futures[future]
-                    self.log.error('error enabling bucket encryption: %s\n%s',
-                                   bucket['Name'], future.exception())
-                results += filter(None, [future.result()])
+                bucket = futures[future]
+                try:
+                    future.result()
+                except Exception as e:
```

Comment:
going back to the original, it just needs a one line continue on error

Reply:
I'm not sure what you mean?

Reply:
it might be that future.exception() is not setup correctly in the original.  The exception in process_bucket was bubbling out of the iteration and never getting here.

Reply:
the exception wasn't bubbling till future.result() was called, the issue is that it should have had a `continue` in the if f.exception(): clause